spec/unit/puppet/provider/puppet_config/cli_spec.rb

```diff
@@ -17,25 +17,47 @@
         end
         context 'when validating instances' do
           before :each do
-            Puppet[:confdir] = '/etc/puppetlabs/puppet'
-            described_class.expects(:puppet).with('config', 'print', 'all', '--confdir', '/etc/puppetlabs/puppet', '--section', 'main').returns \
-'agent_catalog_run_lockfile = /var/lib/puppet/state/agent_catalog_run.lock
-agent_disabled_lockfile = /var/lib/puppet/state/agent_disabled.lock
-allow_duplicate_certs = false
-allow_variables_with_dashes = false
-'
-            described_class.expects(:puppet).with('config', 'print', 'all', '--confdir', '/etc/puppetlabs/puppet', '--section', 'master').returns \
-'agent_catalog_run_lockfile = /var/lib/puppet/state/agent_catalog_run.lock
-agent_disabled_lockfile = /var/lib/puppet/state/agent_disabled.lock
-allow_duplicate_certs = false
-allow_variables_with_dashes = false
-'
-            described_class.expects(:puppet).with('config', 'print', 'all', '--confdir', '/etc/puppetlabs/puppet', '--section', 'agent').returns \
-'agent_catalog_run_lockfile = /var/lib/puppet/state/agent_catalog_run.lock
-agent_disabled_lockfile = /var/lib/puppet/state/agent_disabled.lock
-allow_duplicate_certs = false
-allow_variables_with_dashes = false
-'
+            if #{os} =~ /windows/
```

Comment:
It would be better to use a `context` with an `if` statement, e.g.


```ruby
context 'on Windows', if: os == 'Windows' do

done

context 'on other OSes', if: os != 'Windows' do

done
```